text/0000-rustflags.md

```diff
@@ -0,0 +1,289 @@
+the `profile.rustflags` values.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+There currently exists multiple ways of setting Rust compiler flags when building a Rust project with Cargo. As we mentioned
+earlier, there about 7 different ways that already exist today and this RFC is proposing to add yet another option. This could
+lead to confusion about the best way to set Rust compiler flags in the community.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+## Rationale
+
+This design provides a simple mechanism to specify the set of Rust compiler flags
+a given crate should be built with. This design also has the benefit of not forcing
+all crates in the dependency graph, including upstream dependencies and transitive
+dependencies, to be built in the same manner. As with the given example above, setting
+the rustc flag `-C instrument-coverage` forces the compiler to do an extra amount
+of work to instrument all of the libraries in a given crate. If this flag was passed
+for all transitive dependencies, that would only add to the amount of work that needs
+to be done by the compiler. With this new feature, the rustc flags set via the `--rustflag`
+cargo option would only affect the root crate or the members of the current workspace.
+
+## Alternatives
+
+### Alternative 1: existing RUSTFLAGS manifest keys
+
+Accepting this alternative would mean changes would not be made to either cargo or rustc. The existing behavior
+does allow for a way to set custom Rust compiler flags that would be passed to the compiler at the invocation of rustc
+by cargo. This would still leave Cargo in a place where a Rust developer would not be able to enable/disable rustc flags
+on a crate by crate basis. As with the example of running code coverage for a Rust project, this would mean all libraries
+would be instrumented leading to larger code coverage files and more compilation time spent by the compiler.
+
+### Alternative 2: new build.<crate_name>.rustflags manifest key
+
+This alternative proposes another way to implement such a feature that would allow a Rust developer to specify through cargo
+configuration files exactly which crate enables which Rust compiler flag. There already exists a `rustflags` key under the
+`[build]` section of the cargo configuraiton file and this would add to that existing pattern to enable selective enabling
+rustc flags. An issue with this approach is that changes to the config file is more cumbersome than simply adding commands at
+the CLI. This approach I believe would also make it the feature more complicated in cases where a Rust developer wants to
+enable a specific flag for a set of crates such as the crates within a given workspace. This approach would mean each crate
+would need a separate section in the cargo configuration file to enable said rustc flag.
```

Comment:
An advantage of this approach is extensible - you could have the same effect as this RFC proposes by only supporting `current-crate` as the crate name, then extend it to other names later if needed

Reply:
I'll add this detail to this alternatives section. Thanks for calling that out.